xicam/plugins/__init__.py

```diff
@@ -70,462 +72,352 @@ def load_timer():
     elapser = lambda: end - start
 
 
-class XicamPluginManager(PluginManager):
+class State(Enum):
+    READY = auto()
+    DISCOVERING = auto()
+    LOADING = auto()
+    INSTANTIATING = auto()
+
+
+class Filters(Enum):
```

Comment:
Regarding the `_notify` method, should `Filters` and/or `attach` have documentation about what a filter is used for? For example, the `filter` parameter in `attach` defines if a callback should be invoked or not (by default, filter is None, which means the callback attached will always be called).